File: app/src/main/java/dev/eyosiyas/smsblocker/util/PrefManager.java
package dev.eyosiyas.smsblocker.util;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

public class PrefManager {
    private final SharedPreferences preferences;
    private final Context context;

    public PrefManager(Context context) {
        preferences = PreferenceManager.getDefaultSharedPreferences(context);
        this.context = context;
    }

    public boolean isNuclearEnabled() {
        return preferences.getBoolean(Constant.NUCLEAR, false);
    }

    public void setNuclear(boolean enable) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(Constant.NUCLEAR, enable);
        editor.apply();
    }

    public void setStartsWith(String startsWith) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(Constant.STARTS_WITH, startsWith);
        editor.apply();
    }

    public boolean isStartsWithEnabled() {
        return preferences.getString(Constant.STARTS_WITH, "").length() > 1;
    }

    public String getStartsWith() {
        return preferences.getString(Constant.STARTS_WITH, "");
    }

    public void setEndsWith(String endsWith) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(Constant.ENDS_WITH, endsWith);
        editor.apply();
    }

    public boolean isEndsWithEnabled() {
        return preferences.getString(Constant.ENDS_WITH, "").length() > 1;
    }

    public String getEndsWith() {
        return preferences.getString(Constant.ENDS_WITH, "");
    }
}

File: app/src/main/java/dev/eyosiyas/smsblocker/util/PrefManager.kt
package dev.eyosiyas.smsblocker.util

import android.content.Context

class PrefManager constructor(context: Context) {
    private val preferences = context.getSharedPreferences("Fire", Context.MODE_PRIVATE)
    var startsWith: String?
        get() = preferences.getString(Constant.STARTS_WITH, "")
        set(startsWith) = preferences.edit().putString(Constant.STARTS_WITH, startsWith).apply()

    var endsWith: String?
        get() = preferences.getString(Constant.ENDS_WITH, "")
        set(endsWith) = preferences.edit().putString(Constant.ENDS_WITH, endsWith).apply()

    var firstRun: Boolean
        get() = preferences.getBoolean(Constant.FIRST_RUN, true)
        set(value) = preferences.edit().putBoolean(Constant.FIRST_RUN, value).apply()

    var locale: String
        get() = preferences.getString(Constant.LANGUAGE, "en")!!
        set(locale) {
            preferences.edit().putString(Constant.LANGUAGE, locale).apply()
        }
    var blockingRule: String
        get() = preferences.getString(Constant.BLOCKING_RULE, "")!!
        set(rule) {
            preferences.edit().putString(Constant.BLOCKING_RULE, rule).apply()
        }
}